### PR TITLE
lightning: add error handler for distributed import

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 1]
```diff
@@ -180,3 +180,20 @@ type = "sql"
 另外一个实例的配置修改为只导入 `05001 ~ 10000` 数据文件即可。
 
 其他步骤请参考示例 1 中的相关步骤。
+
+## 错误处理
+
+### 部分 TiDB Lightning 节点异常终止
+
+在并行导入过程中，如果一个或多个 TiDB Lightning 节点异常终止，需要首先根据日志中的报错明确异常退出的原因，然后根据错误类型做不不同处理：
```
Comment (gozssky) [inline tidb-lightning/tidb-lightning-distributed-import.md:188]:
```suggestion
在并行导入过程中，如果一个或多个 TiDB Lightning 节点异常终止，需要首先根据日志中的报错明确异常退出的原因，然后根据错误类型做不同处理：
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 2]
```diff
@@ -180,3 +180,20 @@ type = "sql"
...
+
+1. 如果是非异常型退出(如手动 Kill 等)，或内存溢出被操作系统终止等，可以在适当调整配置后直接重启 TiDB Lightning，无须任何其他操作。
+
+2. 如果是不影响数据正确性的报错，如网络超时等错误，可以在每一个失败的节点上使用 tidb-lightning-ctl 工具清除断点续传源数据中记录的错误，然后直接重启这些异常的节点直接从断点位置继续导入即可，详见 [checkpoint-error-ignore](/tidb-lightning/tidb-lightning-checkpoints#--checkpoint-error-ignore)
```
Comment (gozssky) [inline tidb-lightning/tidb-lightning-distributed-import.md:192]: > 然后直接重启这些异常的节点直接从断点位置继续导入即可

可能需要断个句😂

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 3]
```diff
@@ -180,3 +180,20 @@ type = "sql"
...
+
+3. 如果是影响数据正确性的报错，如 Checksum mismatch, 源文件中有非法的数据等导致的报错，需要在每一个失败的节点上使用 tidb-lightning-ctl 工具清除失败的表中已导入的数据及断点续传相关的源数据，详见 [checkpoint-error-destroy](/tidb-lightning/tidb-lightning-checkpoints#--checkpoint-error-destroy)。由于此命令会删除掉下游导入失败表中的所有已导入的数据，因此在所有 TiDB Lightning 节点上面（包括任务正常结束的）重新配置和导入失败的表对应的数据，此时可以配置 filters 参数只导入报错失败相关的表即可。
+
+### 导入过程中报错 "target table is calculating checksum, please wait unit the checksum is finished and try again"
+
+在部分并行导入的场景，如果的表比较多或者一些需要并行导入的表的数据量比较小，可能会出现当一个或多个任务开始处理某个表的时候，此表对应的其他任务已经完成，并正在校验数据的一致性，此时，由于校验数据一致性的时候，不支持其他数据的写入，因此对应的任务会返回 "target table is calculating checksum, please wait unit the checksum is finished and try again" 错误。遇到这个错误时，可以等当前正在执行校验的任务完成校验后，再启动这些失败的任务即可，不会影响数据的正确性。
```
Comment (gozssky) [inline tidb-lightning/tidb-lightning-distributed-import.md:198]:
```suggestion
在部分并行导入的场景，如果表比较多或者一些需要并行导入的表的数据量比较小，可能会出现当一个或多个任务开始处理某个表的时候，此表对应的其他任务已经完成，并正在校验数据的一致性，此时，由于校验数据一致性的时候，不支持其他数据的写入，因此对应的任务会返回 "target table is calculating checksum, please wait unit the checksum is finished and try again" 错误。遇到这个错误时，可以等当前正在执行校验的任务完成校验后，再启动这些失败的任务即可，不会影响数据的正确性。
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 4]
```diff
@@ -180,3 +180,20 @@ type = "sql"
...
 
 其他步骤请参考示例 1 中的相关步骤。
+
+## 错误处理
+
+### 部分 TiDB Lightning 节点异常终止
+
+在并行导入过程中，如果一个或多个 TiDB Lightning 节点异常终止，需要首先根据日志中的报错明确异常退出的原因，然后根据错误类型做不不同处理：
+
+1. 如果是非异常型退出(如手动 Kill 等)，或内存溢出被操作系统终止等，可以在适当调整配置后直接重启 TiDB Lightning，无须任何其他操作。
```
Comment (shichun-0415) [inline tidb-lightning/tidb-lightning-distributed-import.md:190]:
```suggestion
1. 如果是正常退出(如手动 Kill 等)，或内存溢出被操作系统终止等，可以在适当调整配置后直接重启 TiDB Lightning，无须任何其他操作。
```

File: tidb-lightning/tidb-lightning-distributed-import.md (hunk 2 above)
Comment (shichun-0415) [inline tidb-lightning/tidb-lightning-distributed-import.md:192]:
```suggestion
2. 如果是不影响数据正确性的报错，如网络超时，可以在每一个失败的节点上使用 tidb-lightning-ctl 工具清除断点续传源数据中记录的错误，然后重启这些异常的节点，从断点位置继续导入，详见 [checkpoint-error-ignore](/tidb-lightning/tidb-lightning-checkpoints#--checkpoint-error-ignore)。
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 5]
```diff
@@ -180,3 +180,20 @@ type = "sql"
...
+
+2. 如果是不影响数据正确性的报错，如网络超时等错误，可以在每一个失败的节点上使用 tidb-lightning-ctl 工具清除断点续传源数据中记录的错误，然后直接重启这些异常的节点直接从断点位置继续导入即可，详见 [checkpoint-error-ignore](/tidb-lightning/tidb-lightning-checkpoints#--checkpoint-error-ignore)
+
+3. 如果是影响数据正确性的报错，如 Checksum mismatch, 源文件中有非法的数据等导致的报错，需要在每一个失败的节点上使用 tidb-lightning-ctl 工具清除失败的表中已导入的数据及断点续传相关的源数据，详见 [checkpoint-error-destroy](/tidb-lightning/tidb-lightning-checkpoints#--checkpoint-error-destroy)。由于此命令会删除掉下游导入失败表中的所有已导入的数据，因此在所有 TiDB Lightning 节点上面（包括任务正常结束的）重新配置和导入失败的表对应的数据，此时可以配置 filters 参数只导入报错失败相关的表即可。
```
Comment (shichun-0415) [inline tidb-lightning/tidb-lightning-distributed-import.md:194]:
```suggestion
3. 如果是影响数据正确性的报错，如 checksum mismatched，表示源文件中有非法的数据，则需要在每一个失败的节点上使用 tidb-lightning-ctl 工具，清除失败的表中已导入的数据及断点续传相关的源数据，详见 [checkpoint-error-destroy](/tidb-lightning/tidb-lightning-checkpoints#--checkpoint-error-destroy)。此命令会删除下游导入失败表中已导入的数据，因此，应在所有 TiDB Lightning 节点（包括任务正常结束的）重新配置和导入失败的表的数据，可以配置 filters 参数只导入报错失败的表。
```

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 6]
```diff
@@ -180,3 +180,19 @@ type = "sql"
...
+
+### 导入过程中报错 "target table is calculating checksum, please wait unit the checksum is finished and try again"
+
+在部分并行导入的场景，如果的表比较多或者一些需要并行导入的表的数据量比较小，可能会出现当一个或多个任务开始处理某个表的时候，此表对应的其他任务已经完成，并正在校验数据的一致性，此时，由于校验数据一致性的时候，不支持其他数据的写入，因此对应的任务会返回 "target table is calculating checksum, please wait unit the checksum is finished and try again" 错误。遇到这个错误时，可以等当前正在执行校验的任务完成校验后，再启动这些失败的任务即可，不会影响数据的正确性。
```
Comment (shichun-0415) [inline tidb-lightning/tidb-lightning-distributed-import.md:198]:
```suggestion
在部分并行导入场景，如果表比较多或者一些表的数据量比较小，可能会出现当一个或多个任务开始处理某个表的时候，此表对应的其他任务已经完成，并开始数据一致性校验。此时，由于数据一致性校验不支持写入其他数据，对应的任务会返回 "target table is calculating checksum, please wait unit the checksum is finished and try again" 错误。等校验任务完成，重启这些失败的任务，报错会消失，数据的正确性也不会受影响。
```

**Review by hfxsd — APPROVED** [review]
LGTM

File: tidb-lightning/tidb-lightning-distributed-import.md (hunk 2 above)
Comment (hfxsd) [reply tidb-lightning/tidb-lightning-distributed-import.md]: 改了

**Review by gozssky — APPROVED** [review]

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 7]
```diff
@@ -180,3 +180,19 @@ type = "sql"
...
+
+在并行导入过程中，如果一个或多个 TiDB Lightning 节点异常终止，需要首先根据日志中的报错明确异常退出的原因，然后根据错误类型做不同处理：
+
+1. 如果是正常退出(如手动 Kill 等)，或内存溢出被操作系统终止等，可以在适当调整配置后直接重启 TiDB Lightning，无须任何其他操作。
+
+2. 如果是不影响数据正确性的报错，如网络超时，可以在每一个失败的节点上使用 tidb-lightning-ctl 工具清除断点续传源数据中记录的错误，然后重启这些异常的节点，从断点位置继续导入，详见 [checkpoint-error-ignore](/tidb-lightning/tidb-lightning-checkpoints#--checkpoint-error-ignore)。
+
+3. 如果是影响数据正确性的报错，如 checksum mismatched，表示源文件中有非法的数据，则需要在每一个失败的节点上使用 tidb-lightning-ctl 工具，清除失败的表中已导入的数据及断点续传相关的源数据，详见 [checkpoint-error-destroy](/tidb-lightning/tidb-lightning-checkpoints#--checkpoint-error-destroy)。此命令会删除下游导入失败表中已导入的数据，因此，应在所有 TiDB Lightning 节点（包括任务正常结束的）重新配置和导入失败的表的数据，可以配置 filters 参数只导入报错失败的表。
+
+### 导入过程中报错 "target table is calculating checksum, please wait unit the checksum is finished and try again"
```
Comment (gozssky) [inline tidb-lightning/tidb-lightning-distributed-import.md:196]: `unit` -> `until`, 但 lightning 代码里就有这个 typo 🤔

Comment (hfxsd) [reply tidb-lightning/tidb-lightning-distributed-import.md:196]: 已经反馈给高磊老师了。我先把文档中的改掉。

Comment (hfxsd) [inline tidb-lightning/tidb-lightning-distributed-import.md:196]:
```suggestion
### 导入过程中报错 "target table is calculating checksum, please wait until the checksum is finished and try again"
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: tidb-lightning/tidb-lightning-distributed-import.md [hunk 8]
```diff
@@ -180,3 +180,19 @@ type = "sql"
...
+
+1. 如果是正常退出(如手动 Kill 等)，或内存溢出被操作系统终止等，可以在适当调整配置后直接重启 TiDB Lightning，无须任何其他操作。
+
+2. 如果是不影响数据正确性的报错，如网络超时，可以在每一个失败的节点上使用 tidb-lightning-ctl 工具清除断点续传源数据中记录的错误，然后重启这些异常的节点，从断点位置继续导入，详见 [checkpoint-error-ignore](/tidb-lightning/tidb-lightning-checkpoints.md#--checkpoint-error-ignore)。
+
+3. 如果是影响数据正确性的报错，如 checksum mismatched，表示源文件中有非法的数据，则需要在每一个失败的节点上使用 tidb-lightning-ctl 工具，清除失败的表中已导入的数据及断点续传相关的源数据，详见 [checkpoint-error-destroy](/tidb-lightning/tidb-lightning-checkpoints.md#--checkpoint-error-destroy)。此命令会删除下游导入失败表中已导入的数据，因此，应在所有 TiDB Lightning 节点（包括任务正常结束的）重新配置和导入失败的表的数据，可以配置 filters 参数只导入报错失败的表。
+
+### 导入过程中报错 "Target table is calculating checksum. Please wait until the checksum is finished and try again"
+
+在部分并行导入场景，如果表比较多或者一些表的数据量比较小，可能会出现当一个或多个任务开始处理某个表的时候，此表对应的其他任务已经完成，并开始数据一致性校验。此时，由于数据一致性校验不支持写入其他数据，对应的任务会返回 "target table is calculating checksum, please wait unit the checksum is finished and try again" 错误。等校验任务完成，重启这些失败的任务，报错会消失，数据的正确性也不会受影响。
```
Comment (shichun-0415) [inline tidb-lightning/tidb-lightning-distributed-import.md:198]:
```suggestion
在部分并行导入场景，如果表比较多或者一些表的数据量比较小，可能会出现当一个或多个任务开始处理某个表的时候，此表对应的其他任务已经完成，并开始数据一致性校验。此时，由于数据一致性校验不支持写入其他数据，对应的任务会返回 "Target table is calculating checksum. Please wait until the checksum is finished and try again" 错误。等校验任务完成，重启这些失败的任务，报错会消失，数据的正确性也不会受影响。
```

**Pull request merged**